### ISSUE TITLE
TV Headend servermode

### ISSUE THREAD

**Issue by sualfred** [picons/picons-source#88]
Hey there,

it's not really a issue, but I don't know where I should ask:
I want to rerun my TVH based picon creation, but it only takes care for new mapped channels, even if I delete the output folder.  Any hints? :)

**Comment by mocram** [comment]
None of us are using tvheadend, to be honest I have no idea what you mean...sorry

Try file mode, if that works and there really is a problem in the script for server mode, let me know so I can delete that.

**Comment by sualfred** [comment]
There is no problem with the server mode. It worked great on the first run. 
But I mapped 4 new services recently and it seems that it only creates the picons for these 4 channels.
Is there any check implemented? Maybe the image cache of TVH? (eg if image is already set, skip that channel)

**Comment by mocram** [comment]
What does the line `TvHeadend (server-mode): Converting channel: $channel/$channelcount` say for channelcount?

Do all the channels have picon enabled in the settings? As far as I can see at first glance there is no such thing as a check, it's just requesting all the channels. I however did not implement that piece of code, a user contributed it.

Only thing you could try is just letting it try on your channellist files, to see if those results are comparable...If that works, you could troubleshoot if perhaps the tvheadend api is returning correct results.

**Comment by sualfred** [comment]
The picon setting in TVH is global (eg -> Picon path: file://media/b8632618-935b-44a9-8229-e6102ed6bf86/omv1/tvheadend/snp-full/%C.png for SNP).
TVH will fill all icon paths of the mapped serviced with it. %C is going to replaced with the channel name.

I've noticed that build-serverlist.sh gatheres all channels but the servicelist-tvheadend-servermode-snp is empty.

**Comment by mocram** [comment]
snp is not supported for tvheadend, only srp is/was

**Comment by sualfred** [comment]
Err... it worked once? My ~100 SNP icons did not appeared by magic :)

**Comment by mocram** [comment]
Let me rephrase that. Tvheadend did not have the snp logic implemented.

Could you try srp just as a test

**Comment by sualfred** [comment]
Too fast replied. SRP doesn't work too.

**Comment by mocram** [comment]
Next try config file mode, if that fails asswell, they probably changed someting inside tvheadend...

**Comment by mocram** [comment]
Also what does the api return?

```
http://$TVH_USER:$TVH_PASS@$TVH_HOST:$TVH_PORT/api/channel/grid?start=0&limit=1
```

**Comment by sualfred** [comment]
TVH wasn't updated since weeks.  

{"entries":[{"uuid":"f574ff01e9a644ed14be258786eb3b3a","enabled":true,"autoname":false,"name":"PHOENIX HD","number":70,"icon":"file://media/b8632618-935b-44a9-8229-e6102ed6bf86/omv1/tvheadend/snp/phoenixhd.png","icon_public_url":"imagecache/134","epgauto":true,"epggrab":["7bfaeb73d8e3fa29c10ef94f56f4229f"],"dvr_pre_time":0,"dvr_pst_time":0,"epg_running":-1,"services":["caacbf0323870c80331a3843b0aefff5"],"tags":["bf99ad9c8788ca980f1e9e019e5a11b5","7e1448575c804df1e2da2d4542252689","dd3fe18ee904abcf65d6936fba31b48f"],"bouquet":""}],"total":112}

**Comment by mocram** [comment]
You have the following:

```
"icon":"file://media/b8632618-935b-44a9-8229-e6102ed6bf86/omv1/tvheadend/snp/phoenixhd.png"
```

yet it expects

```
"icon":"picon://1_0_.*_.*_.*_.*_.*_0_0_0"
```

For the channels that do work, it probably returns the expected value.
So there is nothing wrong with the code.

**Comment by mocram** [comment]
double close

**Comment by AbuBaniaz** [comment]
Just for info, the SNP code was added to TVHeadend
```
https://tvheadend.org/issues/3525
```
